packages/core/src/features/st-part.ts

```diff
@@ -0,0 +1,123 @@
+        meta: StylableMeta,
+        node: AST,
+        rule: postcss.Rule
+    ): void {
+        const stPartData = plugableRecord.getUnsafe(meta.data, dataKey);
+        const name = node.value;
+        if (!stPartData[name]) {
+            const feature = node.type === `class` ? CSSClass : CSSType;
+            const symbol = feature.get(meta, name);
+            if (!symbol) {
+                return;
+            }
+            stPartData[name] = {
+                node: rule,
+                symbol,
+            };
+            // deprecated
+            ignoreDeprecationWarn(() => {
+                meta.simpleSelectors[name] = stPartData[name];
+            });
+        }
+    },
+});
+
+// API
+
+export function get(meta: StylableMeta, name: string): StylablePart | undefined {
+    const state = plugableRecord.getUnsafe(meta.data, dataKey);
+    return state[name];
+}
+
+export function resolveAll(meta: StylableMeta, resolver: StylableResolver) {
+    const resolvedClasses: Record<string, Array<CSSResolve<ClassSymbol | ElementSymbol>>> = {};
+    for (const [className, classSymbol] of Object.entries(CSSClass.getAll(meta))) {
+        resolvedClasses[className] = resolver.resolveExtends(
+            meta,
+            className,
+            false,
+            undefined,
+            (res, extend) => {
+                const decl = findRule(meta.ast, '.' + className);
+                if (decl) {
+                    // ToDo: move to STExtends
+                    if (res && res._kind === 'js') {
+                        meta.diagnostics.error(decl, diagnostics.CANNOT_EXTEND_JS(), {
+                            word: decl.value,
+                        });
+                    } else if (res && !res.symbol) {
+                        meta.diagnostics.error(
+                            decl,
+                            diagnostics.CANNOT_EXTEND_UNKNOWN_SYMBOL(extend.name),
+                            { word: decl.value }
+                        );
+                    } else {
+                        meta.diagnostics.error(decl, diagnostics.IMPORT_ISNT_EXTENDABLE(), {
+                            word: decl.value,
+                        });
+                    }
+                } else {
+                    if (classSymbol.alias) {
+                        meta.ast.walkRules(new RegExp('\\.' + className), (rule) => {
+                            meta.diagnostics.error(
+                                rule,
+                                diagnostics.UNKNOWN_IMPORT_ALIAS(className),
+                                { word: className }
+                            );
+                            return false;
+                        });
+                    }
+                }
+            }
+        );
+    }
+
+    const resolvedElements: Record<string, Array<CSSResolve<ClassSymbol | ElementSymbol>>> = {};
+    for (const k of Object.keys(CSSType.getAll(meta))) {
```

Comment:
`k`?

Reply:
it was a direct copy from the transformer, it should have changed to `name`